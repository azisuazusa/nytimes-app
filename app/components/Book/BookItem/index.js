/*
 * @flow
 */
import React, { Component } from 'react';
import { Card, CardItem, Body, Text, Grid, Row } from 'native-base';
import { TouchableOpacity } from 'react-native';

type Props = {
  title: string,
  description: string,
  author: string,
  publishedDate: string,
  productUrl: string,
  onPress: (productUrl: string) => void
};

export default class BookItem extends Component<Props> {
  render() {
    const {
      title,
      description,
      author,
      publishedDate,
      productUrl,
      onPress
    } = this.props;

    return (
      <TouchableOpacity onPress={() => onPress(productUrl)}>
        <Card>
          <CardItem header>
            <Text>{title}</Text>
          </CardItem>
          <CardItem>
            <Body><Text>{description}</Text></Body>
          </CardItem>
          <CardItem footer>
            <Grid>
              <Row><Text>{author}</Text></Row>
              <Row><Text>{publishedDate}</Text></Row>
            </Grid>
          </CardItem>
        </Card>
      </TouchableOpacity>
    );
  }
}
