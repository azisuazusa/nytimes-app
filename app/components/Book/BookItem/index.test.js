import React from 'react';
import renderer from 'react-test-renderer';
import BookItem from '../BookItem';

it('BookItem renders correctly', () => {
  const tree = renderer.create(
    <BookItem
      title={'Sunset Bersama Rosie'}
      description={'Menikmati indahnya sunset bersama Rosie'}
      author={'Tere Liye'}
      publishedDate={'20 Januari 2016'}
      productUrl={'https://azisuazusa.blogspot.com'}
      onPress={() => null} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
