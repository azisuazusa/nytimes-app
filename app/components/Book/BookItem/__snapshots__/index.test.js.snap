// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookItem renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      header={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": null,
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
        uppercase={false}
      >
        Sunset Bersama Rosie
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "stretch",
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          uppercase={false}
        >
          Menikmati indahnya sunset bersama Rosie
        </Text>
      </View>
    </View>
    <View
      footer={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": null,
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
            uppercase={false}
          >
            Tere Liye
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
            uppercase={false}
          >
            20 Januari 2016
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
